tt_metal/impl/dispatch/command_queue.hpp

```diff
@@ -23,6 +22,10 @@
 
 namespace tt::tt_metal {
 
+class Event;
+class Trace;
+using RuntimeArgs = std::vector<std::variant<Buffer*, uint32_t>>;
```

Comment:
for such complex forward declarations i'd prefer a separate header like `metal_fwd.hpp`

Reply:
there's a few of these scattered throughout the code + several aliases (`using` ... ) that might be worthwhile to consolidate, perhaps @tt-dma can look into it in his subsequent PRs